File: box/count-number-of-maximum-bitwise-or-subsets/chatgpt-bitwise-or.py
def countMaxOrSubsets(nums):
    max_or = 0
    count = 0

    # Calcula el máximo valor de OR posible
    for num in nums:
        max_or |= num

    # Función DFS para recorrer los subconjuntos
    def dfs(index, current_or):
        nonlocal count
        if current_or == max_or:
            count += 2 ** (len(nums) - index)
            return

        if index == len(nums):
            return

        # Incluimos el elemento actual en el OR
        dfs(index + 1, current_or | nums[index])

        # No incluimos el elemento actual
        dfs(index + 1, current_or)

    dfs(0, 0)
    return count

nums = [3, 1]
nums = [7, 7, 7, 7, 7, 7, 7, 7]
print(countMaxOrSubsets(nums))  # Salida esperada: 2

File: box/count-number-of-maximum-bitwise-or-subsets/chatgpt-bitwise-or.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var countMaxOrSubsets = function (nums) {
  let maxOr = 0;
  let count = 0;

  // Calcula el máximo valor de OR posible
  for (let num of nums) maxOr |= num;

  // Función DFS para recorrer los subconjuntos
  function dfs(index, currentOr) {
    if (currentOr === maxOr) {
      count += 2 ** (nums.length - index);
      return;
    }

    if (index === nums.length) return;

    // Incluimos el elemento actual en el OR
    dfs(index + 1, currentOr | nums[index]);

    // No incluimos el elemento actual
    dfs(index + 1, currentOr);
  }

  dfs(0, 0);
  return count;
};

console.log([3, 1], countMaxOrSubsets([3, 1])); // Output 2
console.log([2, 2, 2], countMaxOrSubsets([2, 2, 2])); // Output 7
console.log([3, 2, 1, 5], countMaxOrSubsets([3, 2, 1, 5])); // Output 6
console.log([9, 9, 9, 9, 9, 9, 9], countMaxOrSubsets([9, 9, 9, 9, 9, 9, 9])); // Output 127
